pinecone/test_suite.go

```diff
@@ -64,29 +63,36 @@ func (ts *IntegrationTests) SetupSuite() {
 		log.Fatalf("Failed to upsert vectors in SetupSuite: %v", err)
 	}
 
-	ts.sourceTag = "test_source_tag"
-	idxConnSourceTag, err := newIndexConnection(newIndexParameters{
-		additionalMetadata: additionalMetadata,
-		host:               ts.host,
-		namespace:          namespace.String(),
-		sourceTag:          ts.sourceTag})
-	require.NoError(ts.T(), err)
-	ts.idxConnSourceTag = idxConnSourceTag
+	// Create collection for pod index suite
+	if ts.indexType == "pods" {
+		createCollection(ts, ctx)
```

Comment:
This strikes me as a little strange -- the fact that we're using a private function instead of the actual Create Collection API call. 

Since we now have a `collectionName` field on our interface, wouldn't it be easier to just defined a `collectionName` variable here, call the actual `CreateCollection` endpoint as a real user would and set the `collectionName` interface field?

Reply:
(esp since for deleting a collection, we use the actual `DeleteCollection` endpoint as users would

Reply:
Is there an issue with using the private function if it is calling `ts.client.CreateCollection` under the hood and then asserting on it? I can move the code directly into the original, but it didn't feel very different from how we create the indexes for the tests where we're using a public function: https://github.com/pinecone-io/go-pinecone/blob/main/pinecone/test_suite.go#L165

Reply:
No, no issue at all, it just felt inconsistent to me wrt how we do the other things but it's totally fine as-is!